misk/src/main/kotlin/misk/concurrent/ExecutorServiceModule.kt

```diff
@@ -0,0 +1,26 @@
+package misk.concurrent
+
+import com.google.common.util.concurrent.ThreadFactoryBuilder
+import misk.inject.KAbstractModule
+import java.util.concurrent.ExecutorService
+import java.util.concurrent.Executors
+import kotlin.reflect.KClass
+
+class ExecutorServiceModule(
+  private val annotation: KClass<out Annotation>,
+  private val executorService: ExecutorService
+) : KAbstractModule() {
+  override fun configure() {
+    bind<ExecutorService>().annotatedWith(annotation.java).toInstance(executorService)
+  }
+
+  companion object {
+    fun withCachedThreadPool(
```

Comment:
this is awesome for everything EXCEPT cached thread pools are bad and you should never use them

Reply:
Yeah, got it! Will make a PR to use `Executors.newFixedThreadPool(...)` instead. :)